client/me/purchases/purchases-site/index.jsx

```diff
@@ -28,51 +27,26 @@ import { managePurchase } from '../paths';
 import './style.scss';
 
 const PurchasesSite = ( {
-	showHeader = true,
 	getManagePurchaseUrlFor = managePurchase,
 	hasLoadedSite,
 	isPlaceholder,
 	site,
 	siteId,
 	purchases,
 	name,
-	domain,
 	slug,
+	showSite,
 } ) => {
-	let items;
-
 	const isJetpack = ! isPlaceholder && some( purchases, ( purchase ) => isJetpackPlan( purchase ) );
 
 	if ( isPlaceholder ) {
-		items = times( 2, ( index ) => <PurchaseItem isPlaceholder key={ index } /> );
-	} else {
-		items = purchases.map( ( purchase ) => (
-			<PurchaseItem
-				getManagePurchaseUrlFor={ getManagePurchaseUrlFor }
-				key={ purchase.id }
-				slug={ slug }
-				isDisconnectedSite={ ! site }
-				purchase={ purchase }
-				isJetpack={ isJetpack }
-			/>
-		) );
+		return <PurchaseItem isPlaceholder />;
 	}
 
 	return (
 		<div className={ classNames( 'purchases-site', { 'is-placeholder': isPlaceholder } ) }>
 			<QuerySites siteId={ siteId } />
```

Comment:
I'm not sure why `<QuerySites>` and the following `<AsyncLoad>` components are here since the data is being passed in from the parent component. If I remove these, I noticed it stops a weird jump in the order of the products from happening. Does anyone know why we have these here?

Reply:
Is the data being passed from the parent? It looks like this make sure that the current `site` object is available in state. It's then queried in this component's `connect` function and padded directly to `PurchaseItem`.

Reply:
The `AsyncLoad` component is rendering the `product-plan-overlap-notices` block. I don't think it's causing the order of the products to change unless it's doing its own querying.

Reply:
> Is the data being passed from the parent?

It might be other data coming in. The parent has the purchases by site. 

> product-plan-overlap-notices

What exactly are those?

Reply:
> What exactly are those?

No idea :) looks like something related to Jetpack plans/products.

Reply:
Can you describe what happens during the jump?

Reply:
I have a combination of products on WordPress.com and Jetpack. After the initial load, the order of products changes which gives off the appearance that something is jumping.

Reply:
It looks like `state.purchases.data` is set by the `PURCHASES_USER_FETCH_COMPLETED` action then again by `PURCHASES_SITE_FETCH_COMPLETED`.

`client/state/purchases/reducer.js:updatePurchases` might remove some of the purchases from the first call then add them again to the end.